src/test/groovy/com/skalashynski/spring/springboot/service/StudentServiceSpockTest.groovy

```diff
@@ -0,0 +1,208 @@
+package com.skalashynski.spring.springboot.service
+
+import com.skalashynski.spring.springboot.entity.Student
+import com.skalashynski.spring.springboot.exception.ApiException
+import com.skalashynski.spring.springboot.exception.message.ApiExceptionMessage
+import com.skalashynski.spring.springboot.repository.StudentRepository
+import com.skalashynski.spring.springboot.service.impl.StudentServiceImpl
+import spock.lang.Specification
+
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.stream.Stream
+
+class StudentServiceSpockTest extends Specification {
+
+    private static final LocalDateTime CREATED_AT = LocalDateTime.now()
+    private StudentRepository studentRepository = Mock()
+    private StudentService studentService = new StudentServiceImpl(studentRepository)
+
+    private static Stream<Student> provideStudents() {
+        return Stream.of(
+                new Student(Long.MIN_VALUE, "firstName", "lastName",
+                        LocalDate.of(1993, 1, 17), CREATED_AT),
+                new Student(Long.MAX_VALUE, "someName", "lastName",
+                        LocalDate.of(2000, 12, 1), CREATED_AT),
+                new Student(1L, "", "",
+                        LocalDate.of(1980, 5, 28), CREATED_AT),
+                new Student(20L, "Ivan", null,
+                        LocalDate.of(2005, 8, 10), CREATED_AT)
+        )
+    }
```

Comment:
Remove this method and set students to `where` section in each test. Make tests independent from common resource list.

Reply:
Since I used the students stream not as args but as a resource in `given` section, except for the first save test, I put it down in each method in `given` section. That's right?

Reply:
corrected